app/models/project.rb

```diff
@@ -101,6 +101,7 @@ class Project < ApplicationRecord
              includes(:principal)
                .references(:principals)
                .where("#{Principal.table_name}.type='Group' OR " +
+               "#{Principal.table_name}.type='PlaceholderUser' OR " +
                "(#{Principal.table_name}.type='User' AND " +
                "(#{Principal.table_name}.status=#{Principal::STATUSES[:active]} OR " +
                "#{Principal.table_name}.status=#{Principal::STATUSES[:registered]} OR " +
```

Comment:
Now might be a good time to alter this to something like

```
includes(:principal)
  .references(:principals)
  .merge(Principal.not_builtin_but_with_placeholder_users.active_or_registered)
```

Maybe the `not_builtin...` part might even be omitted as it should be Member's responsibility to only take principals of those types and not for project to ensure that.

Because the status is brought up here. What will the status of a `PlaceholderUser` be? At least in the database it will have to have one. Default on db field `status` is 1 so that should work.

Reply:
@wielinde and I talked about an approach which would be more correct but the end result will be the same due to the data that is actually in the database.

```
  base_scope = includes(:principal).references(:principals)

  group_or_placeholders = base_scope.merge(Principal.where(type: [Group.name, PlaceholderUser.name])
  manageable_users = base_scope.merge(Principal.where(type: User.name).active_or_registered)
  group_or_placeholders.or(manageable_users)
```

Reply:
We extracted the conditions to separate scopes of `Member`